### PR TITLE
user authentication

### PR DESCRIPTION
**Pull request by prakashsakari**
This PR has the following
1. User authentication - user can login with test credentials or even create an account and log in
2. Persistent user login.

GIF:
The-Right-Fit-Store_GIF

### CODE REVIEW

File: src/components/Auth/Login.jsx [hunk 1]
```diff
@@ -1,116 +1,99 @@
...
-            </span>
-          </button>
-        </div>
-        <div className={getClassName(display)}>
-          <span>Password must contain atleast 6 characters</span>
-          <span>Password cannot contain any special character</span>
-        </div>
+    const handleLoginClick = () => {
+        if (credentials.email !== "" && credentials.password !== "") {
+            userLogin(credentials.email, credentials.password)
```
Comment (hiteshverma27) [inline src/components/Auth/Login.jsx:27]: try to use ternary operator to make it more readable

**Review by hiteshverma27 — APPROVED** [review]
left a suggestion, the rest looks good. 
- good naming conventions used
- you can make an index for context and reducers

**Review by sashaboi — APPROVED** [review]
Looks amazing ! The design and the folder structure looks perfect, looking forward to future versions of this project.

File: src/components/Auth/Login.jsx (hunk 1 above)
Comment (prakashsakari) [reply src/components/Auth/Login.jsx:27]: I felt this was more readable so i wrote it this way, but thanks for the suggestion.

**Pull request merged**